plugins/oderave/odephysics.h

```diff
@@ -627,17 +671,34 @@ The possible properties that can be set are: ";
             }
         }
 
+        // get frictions before processing collisions.
+        float combinedFriction;
+        if (_usePerSurfaceFrictions) {
+            // we assume that a friction is a property of a link and hence every geom in a link has the same friction value.
```

Comment:
```suggestion
            // we assume that friction is a property of a link and hence every geom in a link has the same friction value.
```
nit. But also, if we assume the link has uniform friction, why not store it with the link?

Reply:
On ODE side friction is defined for a surface, which is linked to geoms, not links (see official [documentation](https://ode.org/wiki/index.php/Manual#Contact_points:~:text=struct%20dContact%20%7B%0A%20%20%20%20%20dSurfaceParameters%20surface%3B%0A%20%20%20%20%20dContactGeom%20geom%3B%0A%20%20%20%20%20dVector3%20fdir1%3B%0A%7D%3B)). Hence, I believe it's better to store the information about friction on the geom level to comply with ODE notation, but use the assumption that a link is made of the same material, so that the friction for all its geoms is the same - this simplifies the code for extracting the friction from geoms.

Reply:
@fedorrrd 

> On ODE side friction is defined for a surface, which is linked to geoms, not links (see official documentation).

Really? I thought ODE can have different friction coefficient for each contact constraint. I think storing friction on geometry is a right choice. Just  the ode plugin in OpenRAVE does not associate ODE's geom to OpenRAVE's geom (instead, associated to OpenRAVE's link).

I think comment should be reworded to clarify this restriction.

Reply:
@yoshikikanemoto @felixvd does this look better?

```
// Friction is a property of a contact surface in ODE which is linked to ODE geoms.
// Hence _friction is stored on the geom level in openRAVE.
// We assume that OpenRAVE link has a uniform material, therefore every geom in a link has the same _friction value.
// Hence we can just take the friction of the first geom GetGeometry(0).
```

Reply:
`We assume that OpenRAVE link has a uniform material` This is misleading. To me, it is just a limitation of the current implementation. 

maybe we should just put a TODO.
```
TODO: Being able to get corresponding OpenRAVE's geometry from ODE's dGeomID so that we can use friction coefficient from the actual geometries that have the collision.
```